packages/editor/src/client/panel/navigationEditPanel.tsx

```diff
@@ -0,0 +1,336 @@
+  useArticleListQuery,
+  ArticleRefFragment
+} from '../api'
+
+import {useTranslation} from 'react-i18next'
+import {generateID, getOperationNameFromDocument} from '../utility'
+import {ListInput, ListValue} from '../atoms/listInput'
+
+export interface NavigationEditPanelProps {
+  id?: string
+
+  onClose?(): void
+  onSave?(navigation: FullNavigationFragment): void
+}
+
+export interface LinkType {
+  id: string[]
+}
+
+export enum LinkTypes {
+  PageNavigationLink = 'Page',
+  ArticleNavigationLink = 'Article',
+  ExternalNavigationLink = 'External'
+}
+
+export interface NavigationLinkTT extends NavigationLinkInput {
+  label: string
+  type: LinkTypes
+}
+
+export interface NavigationLink {
+  label: string
+  type: string
+  articleID?: string
+  pageID?: string
+  url?: string
+}
+
+export function NavigationEditPanel({id, onClose, onSave}: NavigationEditPanelProps) {
+  const [name, setName] = useState('')
+  const [key, setKey] = useState('')
+  const [navigationLinks, setNavigationLinks] = useState<ListValue<NavigationLink>[]>([])
+  const [pages, setPages] = useState<PageRefFragment[]>([])
+  const [articles, setArticles] = useState<ArticleRefFragment[]>([])
+
+  const testLinkTypes: Record<string, string>[] = []
+  for (const [propertyValue, propertyKey] of Object.entries(LinkTypes)) {
+    // dump-variable is a work-around to avoid "unused-variable"-error
+    testLinkTypes.push({label: propertyKey, value: propertyKey, dump: propertyValue})
+  }
+
+  const {data, loading: isLoading, error: loadError} = useNavigationQuery({
+    variables: {id: id!},
+    fetchPolicy: 'network-only',
+    skip: id === undefined
+  })
+
+  const {data: pageData, loading: isLoadingPageData, error: pageDataError} = usePageListQuery({
+    variables: {first: 50},
+    fetchPolicy: 'no-cache'
+  })
+
+  const {
+    data: articleData,
+    loading: isLoadingArticleData,
+    error: articleDataError
+  } = useArticleListQuery({
+    variables: {first: 50},
+    fetchPolicy: 'no-cache'
+  })
+
+  const [createNavigation, {loading: isCreating, error: createError}] = useCreateNavigationMutation(
+    {
+      refetchQueries: [getOperationNameFromDocument(NavigationListDocument)]
+    }
+  )
+
+  const [
+    updateNavigation,
+    {loading: isUpdating, error: updateError}
+  ] = useUpdateNavigationMutation()
+
+  const isDisabled =
+    isLoading ||
+    isCreating ||
+    isUpdating ||
+    isLoadingPageData ||
+    loadError !== undefined ||
+    pageDataError !== undefined ||
+    isLoadingArticleData ||
+    articleDataError !== undefined
+
+  const {t} = useTranslation()
+
+  useEffect(() => {
+    if (data?.navigation) {
+      setName(data.navigation.name)
+      setKey(data.navigation.key)
+      setNavigationLinks(
+        data.navigation.links
+          ? data.navigation.links.map(link => {
+              return {
+                id: generateID(),
+                value: {
+                  label: link.label,
+                  type:
```

Comment:
Why don't you keep the value of `link.__typename`? Why are you replacing it with another string if you already have three unique strings to define the type?